server/spec/mutations/host_nodes/update_spec.rb

```diff
@@ -27,67 +32,115 @@
         }.not_to change{ node.labels }
       end
 
-      it 'notifies grid nodes' do
-        mutation = described_class.new(
-          host_node: node,
-          labels: []
-        )
-        expect(mutation).to receive(:notify_grid).once.with(grid)
-        mutation.run
+      context 'with connected nodes' do
+        let(:node2) { grid.create_node!('node-2', node_id: 'BBB', connected: true) }
+        let(:rpc_client2) { instance_double(RpcClient) }
+
+        before do
+          node.set(connected: true)
+          allow(RpcClient).to receive(:new).with(node2.node_id, Integer).and_return(rpc_client2)
+        end
+
+        it 'notifies both grid nodes' do
+          expect(rpc_client).to receive(:notify).with('/agent/node_info', hash_including(id: 'AA'))
+          expect(rpc_client2).to receive(:notify).with('/agent/node_info', hash_including(id: 'BBB'))
+
+          outcome = described_class.run(
+            host_node: node,
+            labels: []
+          )
+          expect(outcome).to be_success
+        end
       end
     end
 
-    context 'availability' do
-      context 'drain' do
+    describe 'availability' do
+      context 'for an active node' do
+        before do
+          node.set(availability: HostNode::Availability::ACTIVE) # default
+        end
+
         it 'updates availability to drain' do
           expect {
             outcome = described_class.new(
               host_node: node,
               availability: HostNode::Availability::DRAIN,
               labels: []
             ).run
-            expect(outcome.success?).to be_truthy
+            expect(outcome).to be_success
           }.to change{ node.reload.availability }.from(HostNode::Availability::ACTIVE).to(HostNode::Availability::DRAIN)
         end
 
-        it 'stops stateful services when draining' do
-          mutation = described_class.new(
-            host_node: node,
-            availability: HostNode::Availability::DRAIN,
-            labels: []
-          )
-          expect(mutation).to receive(:stop_stateful_services)
+        context 'with a running stateful service instance' do
+          let!(:stack) { grid.stacks.find_by(name: 'null') }
+          let!(:stateful_service) { grid.grid_services.create!(stack: stack, name: 'redis', image_name: 'redis:latest', stateful: true, state: 'running') }
+          let!(:stateful_service_instance1) { stateful_service.grid_service_instances.create!(host_node: node, instance_number: 1, desired_state: 'running') }
+
+          it 'stops stateful services when draining' do
+            expect(rpc_client).to receive(:notify).with('/service_pods/notify_update', 'stop') do
+              expect(stateful_service_instance1.desired_state).to eq 'stopped'
```

Comment:
Is this to validate that the desired state is `stopped` when the notification is sent? Maybe a `reload`   is in order?

Reply:
Yes, sending them in the wrong order would be bad, or not actually saving the `desired_state` change. Reload is good.